### PR TITLE
[ufc] Add additional test cases

### PR DESCRIPTION
**Pull request by schmit**

### CODE REVIEW

File: ufc/tests/test-case-kill-switch-flag.json [hunk 1]
```diff
@@ -13,6 +13,11 @@
       "subjectAttributes": { "email": "bob@example.com", "country": "Canada" },
       "assignment": true
     },
+    {
+      "subjectKey": "barbara",
+      "subjectAttributes": { "email": "barbara@example.com", "country": "canada" },
```
Comment (leoromanovsky) [inline ufc/tests/test-case-kill-switch-flag.json:18]: regex match?

Comment (schmit) [reply ufc/tests/test-case-kill-switch-flag.json:18]: Testing case-sensitivity: this should not match `Canada` (like bob above)

File: ufc/tests/test-case-new-user-onboarding-flag.json [hunk 2]
```diff
@@ -23,6 +23,16 @@
       "subjectAttributes": {"email": "test@test.com", "country": "Mexico", "age": 25},
       "assignment": "blue"
     },
+    {
+      "subjectKey": "zach",
+      "subjectAttributes": {"email": "test@test.com", "country": "Mexico", "age": 25},
+      "assignment": "purple"
+    },
+    {
+      "subjectKey": "Zach",
```
[the remark this answers is not in the document]
Comment (schmit) [reply ufc/tests/test-case-new-user-onboarding-flag.json:32]: Yup case sensitive for the regex match here, and `zach` for the `id`

File: ufc/flags-v1.json [hunk 3]
```diff
@@ -598,9 +598,34 @@
...
       "allocations": [
+        {
+          "key": "id rule",
+          "rules": [
+            {
+              "conditions": [
+                {
+                  "attribute": "id",
+                  "operator": "MATCHES",
+                  "value": "zach"
```
[the remark this answers is not in the document]
Comment (schmit) [reply ufc/flags-v1.json:616]: I totally agree, not quite sure how to accomplish it. I was hoping that "id rule" would be somewhat informative but clearly it wasn't

**Review by aarsilv — COMMENTED** [review]
I'm late to the reviewing party but nice work thanks so much for doing this!

**Pull request merged**